process/interceptors/processor/hdrInterceptorProcessor.go

```diff
@@ -12,9 +13,10 @@ var _ process.InterceptorProcessor = (*HdrInterceptorProcessor)(nil)
 // HdrInterceptorProcessor is the processor used when intercepting headers
 // (shard headers, meta headers) structs which satisfy HeaderHandler interface.
 type HdrInterceptorProcessor struct {
-	headers      dataRetriever.HeadersPool
-	hdrValidator process.HeaderValidator
-	blackList    process.BlackListHandler
+	headers            dataRetriever.HeadersPool
+	hdrValidator       process.HeaderValidator
+	blackList          process.BlackListHandler
+	registeredHandlers []func(topic string, hash []byte, data interface{})
```

Comment:
we need a mutex to protect the array of handlers. Valid for all usages: HdrInterceptorProcessor, MiniblockInterceptorProcessor

Reply:
done